FEM/minkley.cpp

```diff
@@ -0,0 +1,633 @@
+
+    //nothing to do for G_23
+
+    //build G_31
+    Jac.block<6,6>(12,0) = -1./(2.*etaM) * (GM0*smath->P_dev - sigd_curr * dmu_vM.transpose() / etaM);
+
+    //nothing to do for G_32
+
+    //build G_33
+    Jac.block<6,6>(12,12) = 1./dt * smath->ident;
+}
+
+/**************************************************************************
+   FEMLib-Method: Burgers::CaldGdE()
+   Task: Calculates the 12x6 derivative of the residuals with respect to total strain. Implementation fully implicit only.
+   Programing:
+   06/2015 TN Implementation
+**************************************************************************/
+void SolidMinkley::CaldGdE(Eigen::Matrix<double,18,6> &dGdE)
+{
+
+    //Check Dimension of dGdE
+    if (dGdE.cols() != 6 || dGdE.rows() != 18)
+    {
+        std::cout << "WARNING: dGdE given to SolidMinkley::CaldGdE has wrong size. Resizing to 18x6\n";
+        dGdE.resize(18,6);
+    }
+
+    dGdE.setZero(18,6);
+    dGdE.block<6,6>(0,0) = -2. * smath->P_dev - 3. * KM0/GM0 * smath->P_sph;
+}
+
+/**************************************************************************
+   FEMLib-Method: Minkley::DG_DI1()
+   Task: \partial G / \partial I_1
+   Programing:
+   06/2015 TN Implementation
+**************************************************************************/
+double SolidMinkley::DG_DI1(const double alpha)
+{
+    return std::sin(alpha)/3.;
+}
+
+/**************************************************************************
+   FEMLib-Method: Minkley::DG_DJ2()
+   Task: \partial G / \partial J2
+   Programing:
+   06/2015 TN Implementation
+**************************************************************************/
+double SolidMinkley::DG_DJ2(const double theta, const double J2, const double alpha)
+{
+    if (std::abs(theta) < thetaT)
+        return (std::cos(theta) - std::sin(alpha)*std::sin(theta)/std::sqrt(3.))/(2.*std::sqrt(J2));
+    else
+        return (A(theta,alpha) - B(theta,alpha)*std::sin(3.*theta))/(2.*std::sqrt(J2));
+}
+
+/**************************************************************************
+   FEMLib-Method: Minkley::DG_Dtheta()
+   Task: \partial G / \partial theta
+   Programing:
+   06/2015 TN Implementation
+**************************************************************************/
+double SolidMinkley::DG_Dtheta(const double theta, const double J2, const double alpha)
+{
+    if (std::abs(theta) < thetaT)
+        return -std::sqrt(J2) * (std::sin(theta) + std::sin(alpha)*std::cos(theta)/std::sqrt(3.));
+    else
+        return -std::sqrt(J2) * 3. * B(theta,alpha) * std::cos(3.*theta);
+}
+
+/**************************************************************************
+   FEMLib-Method: Minkley::Dtheta_DJ2()
+   Task: \partial theta / \partial J_2
+   Programing:
+   06/2015 TN Implementation
+**************************************************************************/
+double SolidMinkley::Dtheta_DJ2(const double theta, const double J2)
+{
+    return - std::tan(3.*theta)/(2.*J2);
```

Comment:
I am not sure whether it is good to have a space between the value and its sign.